noxfile.py

```diff
@@ -1,36 +1,145 @@
+"""Nox session definitions."""
+
 from __future__ import annotations
 
 import argparse
+import pathlib
+import re
 import shutil
-from pathlib import Path
 
 import nox
 
 
-DIR = Path(__file__).parent.resolve()
-
 nox.needs_version = ">=2024.3.2"
-nox.options.sessions = ["lint", "pylint", "tests"]
+nox.options.sessions = ["lint", "tests"]
 nox.options.default_venv_backend = "uv|virtualenv"
 
 
+ROOT_DIR = pathlib.Path(__file__).parent.resolve()
+DEFAULT_DEV_VENV_PATH = ROOT_DIR / ".venv"
+
+
+def load_from_frozen_requirements(filename: str) -> dict[str, str]:
+    requirements = {}
+    with pathlib.Path(filename).open(encoding="locale") as f:
+        for raw_line in f:
+            if (end := raw_line.find("#")) != -1:
+                raw_line = raw_line[:end]  # noqa: PLW2901 [redefined-loop-name]
+            line = raw_line.strip()
+            if line and not line.startswith("-"):
+                m = re.match(r"^([^=]*)\s*([^;]*)\s*;?\s*(.*)$", line)
+                if m:
+                    requirements[m[1]] = m[2]
+
+    return requirements
+
+
+REQUIREMENTS = load_from_frozen_requirements(ROOT_DIR / "requirements" / "dev.txt")
+
+
 @nox.session
 def lint(session: nox.Session) -> None:
-    """Run the linter."""
+    """Run the linter (pre-commit)."""
     session.install("pre-commit")
     session.run("pre-commit", "run", "--all-files", "--show-diff-on-failure", *session.posargs)
 
 
 @nox.session
 def tests(session: nox.Session) -> None:
     """Run the unit and regular tests."""
-    session.install(".[test]")
+    session.install("-e", ".", "-r", "requirements/dev.txt")
     session.run("pytest", *session.posargs)
 
 
+@nox.session(python=["3.10", "3.11", "3.12"])
+def venv(session: nox.Session) -> None:
+    """
+    Sets up a Python development environment. Use as: `nox -s venv -- [dest_path] [req_preset]
```

Comment:
I think you should swap `req_preset` with `dest_path`, and integrate the requirement in the path of the venv.
Because in most cases you want just to select an environment, but now wou have to specify both.

Reply:
The decision on the order of the arguments was on purpose to force the user to give a non-standard venv a different name. I added the requirements preset to the name of the venv now so the order can be safely changed, but the default name of the generated venvs are now more verbose, so it's not exactly a win-win, but I guess it's ok...